usr/lib/api/policy.c

```diff
@@ -0,0 +1,1728 @@
+            if (rc != CKR_OK)
+                goto out;
+            rc = policy_extract_strength_key(sd, "SYMMETRIC",
+                                             &pp->strengths[i].strength.details.symmetric,
+                                             i);
+            if (rc != CKR_OK)
+                goto out;
+            rc = policy_extract_strength_key(sd, "digest",
+                                             &pp->strengths[i].strength.details.digests,
+                                             i);
+            if (rc != CKR_OK)
+                goto out;
+            rc = policy_extract_strength_key(sd, "signature",
+                                             &pp->strengths[i].strength.details.signatures,
+                                             i);
+            if (rc != CKR_OK)
+                goto out;
+            pp->strengths[i].set = CK_TRUE;
+            rc = policy_check_unmarked(sd->value);
+            if (rc != CKR_OK)
+                goto out;
+        } else {
+            pp->strengths[i].set = CK_FALSE;
+        }
+    }
+ out:
+    if (rc == CKR_OK)
+        rc = policy_check_unmarked(cfg);
+    confignode_deepfree(cfg);
+    return rc;
+}
+
+CK_RV policy_load_policy_cfg(struct policy_private *pp,
+                             FILE *fp, CK_BBOOL *restricting)
+{
+    struct ConfigBaseNode *cfg, *strength, *allowedmechs, *allowedcurves,
+        *allowedmgfs, *allowedkdfs, *allowedprfs;
+    unsigned long reqstrength;
+    CK_RV rc = CKR_OK;
+    unsigned int vers;
+    int i;
+
+    *restricting = CK_FALSE;
+    if (parse_configlib_file(fp, &cfg, parse_error_hook, 0)) {
+        TRACE_ERROR("Parsing policy configuration failed!\n");
+        OCK_SYSLOG(LOG_ERR, "Parsing policy configuration failed!\n");
+        return CKR_FUNCTION_FAILED;
+    }
+    rc = policy_fileversion_check(cfg, "policy-", strlen("policy-"),
+                                  OCK_STRENGTH_MINVERSION, &vers);
+    if (rc != CKR_OK)
+        return rc;
+    strength = confignode_find(cfg, "strength");
+    if (!strength || !confignode_hastype(strength, CT_INTVAL)) {
+        TRACE_ERROR("Invalid strength configuration in policy!\n");
+        OCK_SYSLOG(LOG_ERR, "Invalid strength configuration in policy!\n");
+        rc = CKR_FUNCTION_FAILED;
+        goto out;
+    }
+    strength->flags = 1;
+    reqstrength = confignode_to_intval(strength)->value;
+    for (i = 0; i < NUM_SUPPORTED_STRENGTHS; ++i) {
+        if (reqstrength == supportedstrengths[i]) {
+            TRACE_DEVEL("POLICY: Using strength %lu\n", supportedstrengths[i]);
+            break;
+        } else if (reqstrength > supportedstrengths[i]) {
+            if (i > 0)
+                i -= 1;
+            TRACE_DEVEL("POLICY: Using strength %lu\n", supportedstrengths[i]);
+            break;
+        }
+    }
+    if (i == NUM_SUPPORTED_STRENGTHS && reqstrength > 0) {
+        i -=1;
+        TRACE_DEVEL("POLICY: Using strength %lu\n", supportedstrengths[i]);
```

Comment:
... and if its higher than the highest supported strength, you round down.....

Reply:
done